neuraxle/metaopt/trial.py

```diff
@@ -168,68 +355,80 @@ def is_new_best_score(self):
-
-        :return:
-        """
-        return self.metrics_results['main']['validation_values']
-
-    def get_higher_score_is_better(self):
-        """
-        Return True if higher scores are better for the main metric.
-
-        :return:
-        """
-        return self.metrics_results['main']['higher_score_is_better']
-
     def to_json(self) -> dict:
         """
         Return the trial in a json format.
 
         :return:
         """
         return {
-            'hyperparams': self.hyperparams.to_flat_as_dict_primitive(),
             'status': self.status.value,
             'error': self.error,
             'metric_results': self.metrics_results,
             'error_traceback': self.error_traceback,
+            'start_time': self.start_time.strftime(TRIAL_DATETIME_STR_FORMAT) if self.start_time is not None else '',
+            'end_time': self.end_time.strftime(TRIAL_DATETIME_STR_FORMAT) if self.end_time is not None else '',
+            'split_number': self.split_number
         }
 
     @staticmethod
-    def from_json(trial_json) -> 'Trial':
+    def from_json(trial_json) -> 'TrialSplit':
         """
-        Create a trial object from json.
+        Create a trial split object from json.
 
         :param trial_json: trial json
         :return:
         """
-        return Trial(
-            hyperparams=trial_json['hyperparams'],
-            status=trial_json['status'],
+        return TrialSplit(
+            status=TRIAL_STATUS(trial_json['status']),
             error=trial_json['error'],
+            error_traceback=trial_json['error_traceback'],
             metrics_results=trial_json['metric_results'],
-            error_traceback=trial_json['error_traceback']
+            start_time=datetime.datetime.strptime(trial_json['start_time'], TRIAL_DATETIME_STR_FORMAT),
+            end_time=datetime.datetime.strptime(trial_json['end_time'], TRIAL_DATETIME_STR_FORMAT),
+            split_number=trial_json['split_number']
         )
 
-    def _get_trial_hash(self, hp_dict: Dict):
+    def set_success(self):
         """
-        Hash hyperparams with md5 to create a trial hash.
+        Set trial status to success.
+        """
+        self.status = TRIAL_STATUS.SUCCESS
 
-        :param hp_dict: hyperparams dict
+    def is_success(self):
+        """
+        Set trial status to success.
+        """
+        return self.status == TRIAL_STATUS.SUCCESS
+
+    def set_failed(self, error: Exception):
+        """
+        Set failed trial with exception.
+
+        :param error: catched exception
         :return:
         """
-        current_hyperparameters_hash = hashlib.md5(str.encode(str(hp_dict))).hexdigest()
-        return current_hyperparameters_hash
+        self.status = TRIAL_STATUS.FAILED
+        self.error = str(error)
+        self.error_traceback = traceback.format_exc()
+
+    def set_pipeline(self, pipeline: BaseStep) -> 'TrialSplit':
```

Comment:
This method feels like it might need to be deleted. Same comment for all the other related classes that has the same method.